cql3/cql3_type.cc

```diff
@@ -450,7 +450,7 @@ sstring maybe_quote(const sstring& identifier) {
         // So we can use any of them, for example cident.
         try {
             cql3::util::do_with_parser(identifier, std::mem_fn(&cql3_parser::CqlParser::cident));
-            return identifier;
+            return sstring{identifier};
```

Comment:
string_view changes are nice but isn't this pessimising common case of string not needing a quotes by introducing an allocation?

Reply:
Here we'd need to allocate memory in the previous version too. We'll just use a different constructor of `sstring`

Reply:
An allocation is trivial compared to calling the cql parser.

This is a metadata path. I don't want to encourage anyone to waste anything, but also don't optimize very low frequency functions.

Reply:
ok